drivers/led/is31fl3194/is31fl3194_reg.h

```diff
@@ -0,0 +1,483 @@
+/*
+ * Copyright (c) 2022 Benjamin Björnsson <benjamin.bjornsson@gmail.com>.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_LED_IS31FL3194_H_
+#define ZEPHYR_INCLUDE_DRIVERS_LED_IS31FL3194_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+struct is31fl3194_bitwise {
+	uint8_t bit0 : 1;
+	uint8_t bit1 : 1;
+	uint8_t bit2 : 1;
+	uint8_t bit3 : 1;
+	uint8_t bit4 : 1;
+	uint8_t bit5 : 1;
+	uint8_t bit6 : 1;
+	uint8_t bit7 : 1;
+};
+
+#define IS31FL3194_NUM_LEDS_MAX 3
+#define IS31FL3194_NUM_PATTERNS_MAX 3
+#define IS31FL3194_NUM_COLORS_MAX 3
+
+#define IS31FL3194_ID 0xCEU
+
+#define IS31FL3194_PRODUCT_ID 0x00U
+#define IS31FL3194_OPERATING_CONF 0x01U
+struct is31fl3194_ops_conf {
+	uint8_t ssd : 1;
+	uint8_t rgb : 2;
+	uint8_t unused_01 : 1;
+	uint8_t out : 3;
+	uint8_t unused_02 : 1;
+};
+
+#define IS31FL3194_OUTPUT_ENABLE 0x02U
+struct is31fl3194_out_en {
+	uint8_t en : 3;
+	uint8_t unused_01 : 5;
+};
+
+#define IS31FL3194_CURRENT_BAND 0x03U
+struct is31fl3194_cbx {
+	uint8_t cb1 : 2;
+	uint8_t cb2 : 2;
+	uint8_t cb3 : 2;
+	uint8_t unused_01 : 2;
+};
+
+#define IS31FL3194_HOLD_FUNCTION 0x04U
+struct is31fl3194_hfx {
+	uint8_t ht1 : 1;
+	uint8_t hf1 : 1;
+	uint8_t ht2 : 1;
+	uint8_t hf2 : 1;
+	uint8_t ht3 : 1;
+	uint8_t hf3 : 1;
+	uint8_t unused_01 : 2;
+};
+
+#define IS31FL3194_P1_STATE 0x0DU
+#define IS31FL3194_P2_STATE 0x0EU
+#define IS31FL3194_P3_STATE 0x0FU
+struct is31fl3194_px_state {
+	uint8_t txx : 3;
+	uint8_t unused_01 : 1;
+	uint8_t cs1x : 1;
+	uint8_t cs2x : 1;
+	uint8_t cs3x : 1;
+	uint8_t psx : 1;
+};
+
+#define IS31FL3194_OUT1_CURRENT_LEVEL 0x10U
+#define IS31FL3194_OUT2_CURRENT_LEVEL 0x21U
+#define IS31FL3194_OUT3_CURRENT_LEVEL 0x32U
+struct is31fl3194_outx_cl {
+	uint8_t cl : 8;
+};
+
+#define IS31FL3194_COLOR_1_P1_BASE 0x10U
+#define IS31FL3194_COLOR_1_P2_BASE 0x20U
+#define IS31FL3194_COLOR_1_P3_BASE 0x30U
+#define IS31FL3194_COLOR_2_P1_BASE 0x13U
+#define IS31FL3194_COLOR_2_P2_BASE 0x23U
+#define IS31FL3194_COLOR_2_P3_BASE 0x33U
+#define IS31FL3194_COLOR_3_P1_BASE 0x16U
+#define IS31FL3194_COLOR_3_P2_BASE 0x26U
+#define IS31FL3194_COLOR_3_P3_BASE 0x36U
+struct is31fl3194_px_coly_oz {
+	uint8_t pxcoly_oz : 8;
```

Comment:
I think you can remove the `: 8` here :)

Reply:
I think it's quite nice to state explicitly that we're using the whole `uint8_t` here and IMO it makes sense to be consistent across structs.

Reply:
OK